### PR TITLE
Dev jake

### PR DESCRIPTION
**Pull request by jakeheft**
This PR will:
- Add count_of_games_by_season, count_of_teams, highest_scoring_visitor, highest_scoring_home_team, team_info, most_goals_scored, and fewest_goals_scored
- All method updated in stat_tracker with proper calls to managers
- All game_teams_manager tests pass

### CODE REVIEW

File: lib/game_teams_manager.rb [hunk 1]
```diff
@@ -26,39 +27,12 @@ def total_game_teams(filtered_game_teams = @game_teams)
...
-  end
-
-  def home_games_by_team
-    home_games.group_by do |game_team|
-      game_team.team_id
+      game.hoa == hoa
     end
   end
 
-  def highest_scoring_visitor
```
Comment (dunlapww) [inline lib/game_teams_manager.rb:61]: Jake, have you confirmed this is passing spec harness?

**Review by dunlapww — COMMENTED** [review]
Hey Jake, look good Jake, just curious whether highest scoring home/visitor are passing spec

File: lib/game_teams_manager.rb (hunk 1 above)
Comment (jakeheft) [reply lib/game_teams_manager.rb:61]: Yes, all 4 of highest/lowest visitor/home pass spec harness

**Review by dunlapww — CHANGES_REQUESTED** [review]
Hey Jake, why don't you repull and push to capture Hope's changes then we'll look at this one

**Comment by jakeheft** [comment]
Will, I have now pulled her branch and merged it all. All tests pass and spec harness only has 8 failures now.

**Review by dunlapww — APPROVED** [review]

**Review by dunlapww — APPROVED** [review]

File: lib/games_manager.rb [hunk 2]
```diff
@@ -119,37 +119,18 @@ def season_win_percentage(team_id, season)
...
-          win_percentages_by_season[team_id] = {season =>
-            season_win_percentage(team_id, season)}
-        else
-          win_percentages_by_season[team_id][season] =
-          season_win_percentage(team_id, season)
-        end
+    @stat_tracker.fetch_all_team_ids.reduce({}) do |data, team_id|
+      data[team_id] = all_seasons.reduce({}) do |collector, season|
+        collector[season] = season_win_percentage(team_id, season)
+        collector
```
Comment (jdcarey128) [inline lib/games_manager.rb:130]: Looks much better than what I had. Thanks for refactoring!

File: lib/games_manager.rb [hunk 3]
```diff
@@ -119,37 +119,18 @@ def season_win_percentage(team_id, season)
...
-  # But, should be moved to GamesManger
   def all_seasons
-    unique_seasons = []
-    @games.each do |game|
-      if !unique_seasons.include?(game.season)
-        unique_seasons << game.season
-      end
-    end
-    unique_seasons.sort
+    season_group.keys.sort
```
Comment (jdcarey128) [inline lib/games_manager.rb:123]: Much better.

File: lib/game_teams_manager.rb [hunk 4]
```diff
@@ -123,18 +91,72 @@ def fewest_tackles(season)
...
+  end
+
+  def team_goals_by_game(team_id)
+    games_by_team(team_id).map do |game|
+      game.goals
+    end
+  end
+
+  def most_goals_scored(team_id)
+    team_goals_by_game(team_id).max.to_i
```
Comment (jdcarey128) [inline lib/game_teams_manager.rb:107]: Should team_goals_by_game return an integer already?

**Review by jdcarey128 — APPROVED** [review]

**Review by hopesgit — APPROVED** [review]

**Pull request merged**